component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java

```diff
@@ -126,15 +129,26 @@ protected void initiateAuthenticationRequest(HttpServletRequest request,
             throws AuthenticationFailedException {
         try {
             String username = null;
+            String email = null;
+            boolean isEmailOTPMandatory;
             AuthenticatedUser authenticatedUser;
-            Map<String, String> authenticatorProperties = context.getAuthenticatorProperties();
             Map<String, String> emailOTPParameters = getAuthenticatorConfig().getParameterMap();
             String tenantDomain = context.getTenantDomain();
             context.setProperty(EmailOTPAuthenticatorConstants.AUTHENTICATION, EmailOTPAuthenticatorConstants.AUTHENTICATOR_NAME);
             if (!tenantDomain.equals(EmailOTPAuthenticatorConstants.SUPER_TENANT)) {
                 IdentityHelperUtil.loadApplicationAuthenticationXMLFromRegistry(context, getName(), tenantDomain);
+                Object getPropertiesFromLocal = context.getProperty(IdentityHelperConstants.GET_PROPERTY_FROM_REGISTRY);
+                if (getPropertiesFromLocal == null) {
+                    isEmailOTPMandatory = Boolean.parseBoolean(String.valueOf(context.getProperty
+                            (EmailOTPAuthenticatorConstants.IS_EMAILOTP_MANDATORY)));
+                } else {
+                    isEmailOTPMandatory = Boolean.parseBoolean(emailOTPParameters
+                            .get(EmailOTPAuthenticatorConstants.IS_EMAILOTP_MANDATORY));
+                }
+            } else {
+                isEmailOTPMandatory = Boolean.parseBoolean(emailOTPParameters
```

Comment:
Here we check the tenant, if it is super tenant, then get the parameters from the local. if it is tenant, then get the parameters from context if the file upload in registry, if the file not in registry try to get it from local.